### PR TITLE
Emit metrics from the filesystem actor

### PR DESCRIPTION
**Pull request by dominiklohmann**
This adds a set of metrics to the filesystem actor. The PR is best reviewed commit-by-commit.

### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [x] All user-facing changes have changelog entries
- [x] User-facing changes are reflected on [vast.io](https://github.com/tenzir/vast/tree/master/web)

### CODE REVIEW

File: libvast/src/system/posix_filesystem.cpp [hunk 1]
```diff
@@ -166,12 +170,41 @@ posix_filesystem(filesystem_actor::stateful_pointer<posix_filesystem_state> self
         add_stats("writes", self->state.stats.writes);
         add_stats("reads", self->state.stats.reads);
         add_stats("mmaps", self->state.stats.mmaps);
+        // TODO: this should be called "deletes" or "erasures".
```
Comment (dominiklohmann) [inline libvast/src/system/posix_filesystem.cpp:173]: Note to the reviewer: That'd be a breaking change, so I only left this comment.

File: libvast/src/system/node.cpp [hunk 2]
```diff
@@ -613,15 +613,35 @@ node_state::spawn_command(const invocation& inv,
...
+      auto err = register_component(
+        self, caf::actor_cast<caf::actor>(accountant), "accountant");
+      VAST_ASSERT(err
+                  == caf::none); // Registration cannot fail; empty registry.
+    }
+  }
   // Initialize the file system with the node directory as root.
-  auto fs = self->spawn<caf::detached>(posix_filesystem, self->state.dir);
+  auto fs
+    = self->spawn<caf::detached>(posix_filesystem, self->state.dir, accountant);
```
Comment (patszt) [inline libvast/src/system/node.cpp:644]: accountant can be moved, but it's not a big deal i guess

**Review by patszt — APPROVED** [review]

File: libvast/src/system/node.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/src/system/node.cpp]: Good spot. I made that change while I moved the code over.

File: libvast/src/system/node.cpp [hunk 3]
```diff
@@ -613,15 +613,35 @@ node_state::spawn_command(const invocation& inv,
...
+      VAST_ERROR("{} failed to parse metrics configuration: {}", *self,
+                 cfg.error());
+    } else {
+      accountant = self->spawn(vast::system::accountant, std::move(*cfg),
+                               self->state.dir);
+      auto err = register_component(
+        self, caf::actor_cast<caf::actor>(accountant), "accountant");
+      VAST_ASSERT(err
+                  == caf::none); // Registration cannot fail; empty registry.
+    }
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/src/system/node.cpp]: Moved it, thanks!

File: libvast/src/system/node.cpp [hunk 4]
```diff
@@ -256,6 +256,30 @@ deregister_component(node_actor::stateful_pointer<node_state> self,
...
+    } else {
+      accountant = self->spawn(vast::system::accountant, std::move(*cfg),
+                               self->state.dir);
+      auto err = register_component(
+        self, caf::actor_cast<caf::actor>(accountant), "accountant");
+      // Registration cannot fail; empty registry.
+      VAST_ASSERT(err == caf::none);
+    }
+  }
+  return accountant;
```
Comment (patszt) [inline libvast/src/system/node.cpp:280]:
```suggestion
if (!caf::get_or(content(self->system().config()), "vast.enable-metrics",
                   false))
    return {};
  const auto metrics_opts = caf::get_or(content(self->system().config()),
                                        "vast.metrics", caf::settings{});
  auto cfg = to_accountant_config(metrics_opts);
  if (!cfg) {
    VAST_ERROR("{} failed to parse metrics configuration: {}", *self,
               cfg.error());
    return {};
  }
  auto accountant
    = self->spawn(vast::system::accountant, std::move(*cfg), self->state.dir);
  auto err = register_component(self, caf::actor_cast<caf::actor>(accountant),
                                "accountant");
  // Registration cannot fail; empty registry.
  VAST_ASSERT(err == caf::none);
  return accountant;
```

Simplification with early returns

**Review by patszt — APPROVED** [review]

**Pull request merged**